File: src/main/java/org/jukeboxmc/network/handler/PlayerSkinHandler.java
package org.jukeboxmc.network.handler;

import org.cloudburstmc.protocol.bedrock.packet.PlayerSkinPacket;
import org.jukeboxmc.Server;
import org.jukeboxmc.player.Player;
import org.jukeboxmc.player.skin.Skin;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerSkinHandler implements PacketHandler<PlayerSkinPacket> {

    @Override
    public void handle( PlayerSkinPacket packet, Server server, Player player ) {
        player.setSkin( Skin.fromNetwork( packet.getSkin() ) );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/network/handler/PlayerSkinHandler.kt
package org.jukeboxmc.server.network.handler

import org.cloudburstmc.protocol.bedrock.packet.PlayerSkinPacket
import org.jukeboxmc.api.player.skin.Skin
import org.jukeboxmc.server.JukeboxServer
import org.jukeboxmc.server.extensions.fromNetwork
import org.jukeboxmc.server.player.JukeboxPlayer

class PlayerSkinHandler : PacketHandler<PlayerSkinPacket> {

    override fun handle(packet: PlayerSkinPacket, server: JukeboxServer, player: JukeboxPlayer) {
        player.setSkin(Skin().fromNetwork(packet.skin))
    }
}